apluslms_roman/builder.py

```diff
@@ -10,18 +10,24 @@
 
 class Builder:
     def __init__(self, engine, config, observer=None):
-        assert isdir(config.__path__), "Course configuration path doesn't exists"
+        if not isdir(config.dir):
+            raise ValueError("config.dir isn't a directory.")
         self.config = config
-        self.path = config.__path__
+        self.path = config.dir
         self._engine = engine
         self._observer = observer or StreamObserver()
 
-    def build(self):
+    def buildSteps(self):
+        return [BuildStep.from_config(i, step) for i, step in enumerate(self.config.steps)]
+
+    def build(self, step_list=[]):
         backend = self._engine.backend
         observer = self._observer
-        steps = [BuildStep.from_config(step) for step in self.config.steps]
+        steps = self.buildSteps()
+        if (step_list):
+            name_dict = {step.name: step for step in steps}
+            steps = [name_dict[step] if step in name_dict.keys() else steps[int(step)] for step in step_list]
```

Comment:
This probably should first check if the `step` (probably should be called `ref` here) is number and convert it to integer and then we can raise correct errors.

e.g.
```
step_list = [int(ref) of ref.isdigit() else ref for ref in step_list]
try:
  steps = [steps[ref] if isinstance(ref, int) else name_dict[ref] for ref in step_list]
except KeyError as e:
  # no step with name ref
except IndexError as e:
  # ref is out of step list
```

Finally, we probably should check if there is duplicates. Probably should log a warning that some step is multiple times. Or we could just drop duplicates. `OrderedDict.from_keys(steps).keys()` will yield all steps only once in same order.

Reply:
AttributeError: type object 'collections.OrderedDict' has no attribute 'from_keys'

Reply:
Forgot that with these types there isn't separator, so `list(OrderedDict.fromkeys(steps))`.  `.keys()` is not actually required as iterator over dictionary does the same and we need the `list` there with `keys()` too.

For reference, conversation about this topic: https://stackoverflow.com/questions/480214/how-do-you-remove-duplicates-from-a-list-whilst-preserving-order